### DISCUSSION TITLE
Encode base64 logo

### DISCUSSION THREAD

**Discussion by robin-gustafsson** [covidpass-org/covidpass#134] [Q&A]
What is the best way to base64 the logo to the CovidPass? I have converted my PNG to base64 but when I generating the CovidPass I got "Covid.pkpass, could not be opened."

<img width="1060" alt="Skärmavbild 2021-12-06 kl  23 56 23" src="https://user-images.githubusercontent.com/31063399/144936057-7bec7df2-5a8b-4cd9-9f38-5226584d66fe.png">

We are using https://github.com/covidpass-org/CovidPassApiNet as backend, do we need to hash the image here to? 

<img width="526" alt="Skärmavbild 2021-12-06 kl  23 57 52" src="https://user-images.githubusercontent.com/31063399/144936213-58979a76-28f9-4f2a-a309-70633babb951.png">

<img width="328" alt="Skärmavbild 2021-12-06 kl  23 54 54" src="https://user-images.githubusercontent.com/31063399/144935906-65f508a5-8134-4077-b28a-008251f283a9.png">

**Comment by hauketoenjes** [comment]
I think the hashes have to be the same on both server and client because only a hash is sent to the server to sign it. If the server returns a signed hash that doesn't fit to the pass on the client, the pass is invalid. And the icons are used within the hashing.

**Reply by robin-gustafsson** [reply]
Thanks! 

What type of hash is on the server side? 

Is the hash "9dcdd385e848610f020cf3bfc65ddc413beb5e87" on server side based on the base64 on the client side?

**Reply by marvinsxtr** [reply]
I think the hash is generated from the actual icon.png file inside the zip folder. Basically the manifest.json contains the hashes of all other files in the folder. On the server, we just check for CovidPass icon's hash to prevent generation of arbitrary passes.
